resources/dynamodb_tables.go

```diff
@@ -0,0 +1,714 @@
+	}
+	return nil
+}
+func resolveDynamodbTableGlobalSecondaryIndexKeySchema(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	r := resource.Item.(types.GlobalSecondaryIndexDescription)
+	return resource.Set(c.Name, marshalKeySchema(r.KeySchema))
+}
+func fetchDynamodbTableLocalSecondaryIndexes(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan interface{}) error {
+	p := parent.Item.(*types.TableDescription)
+	for i := range p.LocalSecondaryIndexes {
+		res <- p.LocalSecondaryIndexes[i]
+	}
+	return nil
+}
+func resolveDynamodbTableLocalSecondaryIndexKeySchema(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	r := resource.Item.(types.LocalSecondaryIndexDescription)
+	return resource.Set(c.Name, marshalKeySchema(r.KeySchema))
+}
+func fetchDynamodbTableReplicas(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan interface{}) error {
+	p := parent.Item.(*types.TableDescription)
+	for i := range p.Replicas {
+		res <- p.Replicas[i]
+	}
+	return nil
+}
+func resolveDynamodbTableReplicaGlobalSecondaryIndexes(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	r := resource.Item.(types.ReplicaDescription)
+	if len(r.GlobalSecondaryIndexes) == 0 {
+		return nil
+	}
+
+	val := make([]map[string]interface{}, len(r.GlobalSecondaryIndexes))
+	for i := range r.GlobalSecondaryIndexes {
+		val[i] = map[string]interface{}{
+			"name":                            r.GlobalSecondaryIndexes[i].IndexName,
+			"provisioned_throughput_override": r.GlobalSecondaryIndexes[i].ProvisionedThroughputOverride,
+		}
+	}
+	b, _ := json.Marshal(val)
+	return resource.Set(c.Name, b)
+}
+func fetchDynamodbTableReplicaAutoScalings(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan interface{}) error {
+	par, ok := parent.Item.(*types.TableDescription)
+	if !ok {
+		return fmt.Errorf("expected *types.TableDescription but got %T", parent.Item)
+	}
+
+	c := meta.(*client.Client)
+	svc := c.Services().DynamoDB
+
+	output, err := svc.DescribeTableReplicaAutoScaling(ctx, &dynamodb.DescribeTableReplicaAutoScalingInput{
+		TableName: par.TableName,
+	}, func(o *dynamodb.Options) {
+		o.Region = c.Region
+	})
+	if err != nil {
+		return err
+	}
+
+	for i := range output.TableAutoScalingDescription.Replicas {
+		res <- output.TableAutoScalingDescription.Replicas[i]
+	}
+	return nil
+}
+func resolveDynamodbTableReplicaAutoScalingGlobalSecondaryIndexes(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	r := resource.Item.(types.ReplicaAutoScalingDescription)
+	if len(r.GlobalSecondaryIndexes) == 0 {
+		return nil
+	}
+
+	val := make([]simplifiedGlobalSecondaryIndex, len(r.GlobalSecondaryIndexes))
+	for i := range r.GlobalSecondaryIndexes {
+		d := r.GlobalSecondaryIndexes[i]
+		val[i] = simplifiedGlobalSecondaryIndex{
```

Comment:
Question: Keep this or make it a map of Name vs. simplifiedGlobalSecondaryIndex? Order shouldn't matter anyway.

Reply:
ditto. map could be more useful.